epcpy/utils/parsers.py

```diff
@@ -23,10 +23,54 @@
 from epcpy.epc_schemes.upui import UPUI
 from epcpy.epc_schemes.usdod import USDOD
 from epcpy.utils.common import ConvertException, base64_to_hex, hex_to_binary
-from epcpy.utils.regex import TAG_URI
-
+from epcpy.utils.regex import (
+    BINARY_HEADERS,
+    CPI_GS1_ELEMENT_STRING,
+    EPC_URI,
+    GDTI_GS1_ELEMENT_STRING,
+    GIAI_GS1_ELEMENT_STRING,
+    GINC_GS1_ELEMENT_STRING,
+    GRAI_GS1_ELEMENT_STRING,
+    GS1_ELEMENT_STRING,
+    GSIN_GS1_ELEMENT_STRING,
+    GSRN_GS1_ELEMENT_STRING,
+    GSRNP_GS1_ELEMENT_STRING,
+    HEX_HEADERS,
+    IDPAT_URI,
+    ITIP_GS1_ELEMENT_STRING,
+    PGLN_GS1_ELEMENT_STRING,
+    SGCN_GS1_ELEMENT_STRING,
+    SGLN_GS1_ELEMENT_STRING,
+    SGTIN_GS1_ELEMENT_STRING,
+    SSCC_GS1_ELEMENT_STRING,
+    TAG_URI,
+    UPUI_GS1_ELEMENT_STRING,
+)
+
+EPC_URI_REGEX = re.compile(EPC_URI)
+GS1_ELEMENT_STRING_REGEX = re.compile(GS1_ELEMENT_STRING)
+IDPAT_URI_REGEX = re.compile(IDPAT_URI)
 TAG_URI_REGEX = re.compile(TAG_URI)
-
+BINARY_HEADERS_REGEX = re.compile(BINARY_HEADERS)
+HEX_HEADERS_REGEX = re.compile(HEX_HEADERS)
+
+GS1_ELEMENT_STRING_REGEX_TO_SCHEME: Union[re.Pattern, GS1Element] = {
+    re.compile(SGTIN_GS1_ELEMENT_STRING): SGTIN,
```

Comment:
We could exploit and improve our modeling here: give every GS1Element class a `GS1_ELEMENT_STRING`. Just like we did with binary headers. Then this is just a list comprehension over the GS1Elements classes.

Reply:
Agreed